plugin/storage/es/factory.go

```diff
@@ -167,7 +166,10 @@ func createSpanWriter(
 		return nil, err
 	}
 
-	spanMapping, serviceMapping := GetSpanServiceMappings(cfg.GetNumShards(), cfg.GetNumReplicas(), client.GetVersion())
+	spanMapping, serviceMapping, err := GetSpanServiceMappings(es.TextTemplateBuilder{}, cfg.GetNumShards(), cfg.GetNumReplicas(), client.GetVersion(), cfg.GetIndexPrefix(), cfg.GetUseILM())
```

Comment:
you may also be able to use that struct directly to pass values to `tmpl.Execute()`, instead of declaring an inner struct

Reply:
lastly, might want to move all this mapping-related business into `mappings` package, instead of being in the `factory`

Reply:
Makes sense to move out all the mappings related funcs to `mappings` package. Will work on it, would need sometime though :)

Also, I am afraid now of rebasing/merging with master again - which is causing DCO issues. @albertteoh, would take your few mins to explain what I am doing while rebasing - please help to rectify, if I am doing something wrong.

Reply:
We could do the move in a follow-up PR.

There are two ways to keep up to date with master - rebasing and merging. When you're not changing files affected by other people, rebasing is usually a non-event and has no impact on DCO: `git rebase master`. You have to force-push after that.

Merging master can be done directly from GitHub and also doesn't affect DCO (just remember to do a pull).

The time when DCO gets messed up is when you do either one of those and you have a conflict that you need to resolve. Git will not insert the sign off comment in that case, you need to do that manually.

Reply:
Yep, this mess up did happened twice and due to merge conflicts only. 

It would be great if I can take this move in follow up PR. This has been a biggish PR already :)

Reply:
Will implement all other feedbacks as part of this. Just this refactoring, would do it in another PR - is it fine @yurishkuro ?

Reply:
ok

Reply:
Opened a new issue to track this - 
https://github.com/jaegertracing/jaeger/issues/2813